keps/sig-storage/2485-read-write-once-pod-pv-access-mode/README.md

```diff
@@ -892,13 +1041,25 @@ high level (needs more precise definitions) those may be things like:
   - 99,9% of /health requests per day finish with 200 code
 -->
 
+Defining an SLO for this metric is difficult because a pod may be
+"Unschedulable" due to user error; they mistakenly scheduled a workload that has
+volume conflicts. Additionally, this metric captures volume conflicts for some
+legacy in-tree drivers that do not support the ReadWriteOncePod feature but are
+part of the same scheduler plugin.
+
+Any unexpected increases in
+`scheduler_unschedulable_pods{plugin="VolumeRestrictions"}` should be
```

Comment:
This covers all scheduling related errors? Does it also cover any kubelet/sidecar/CSI related errors? Are there additional metrics we should include here for those types of errors?

Reply:
This metric only covers scheduling errors.

[Kubelet](https://github.com/kubernetes/kubernetes/blob/v1.26.1/pkg/kubelet/metrics/metrics.go) does not seem to have any metrics that might indicate issues.

[CSI sidecars](https://github.com/kubernetes-csi/csi-lib-utils/blob/v0.12.0/metrics/metrics.go) have metrics that might be useful but could also be noisy. For example, if a user enables the feature gate but does not update their CSI sidecars they may see an increase in `csi_sidecar_operations_seconds_bucket{method_name="/csi.v1.Controller/CreateVolume"}`, however I am unable to repro this, all metrics are reporting with label `grpc_status_code="OK"` when I think we would want to match the inverse of this.

Reply:
kubelet has "storage_operation" metrics: https://github.com/kubernetes/kubernetes/blob/336934ab0a94cb9df2b91a6f5216580dfc610e7c/pkg/volume/util/metrics.go#L48

It looks like enforcement is done during the "MountVolume" operation

Reply:
Re: failing with an older sidecar

It looks like the old logic returned nil for unrecognized access modes. So whether or not it fails depends on if the CSI driver implementation was able to handle that or not.
https://github.com/kubernetes-csi/external-provisioner/pull/630/files#diff-243ebed2765f75e6a54f57167212fefb08c3b2a85967ad2acbc0eb78919019c1L469

Reply:
Updated to include kubelet and CSI sidecar metrics that may provide additional signal